### PR TITLE
MPDX-7017: Make Contacts Responsive

### PR DESCRIPTION
**Pull request by OzzieOrca**
I think I got everything working. Feel free to play with the preview and let me know what you think, if you see any way to make this simpler, or see any issues with it. The CSS is a little complex.

I haven't made any attempt to make the content inside each of the 3 panels responsive yet. There are some horizontal scroll bars inside the panels still. That can be another ticket.

John made this proof of concept which was helpful :)
https://codepen.io/j2trumpet/pen/dd1ead3747b95b897e696c6ecf7ef6d2?editors=0100

### CODE REVIEW

File: src/components/Layouts/SidePanelsLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      background: theme.palette.common.white,
+      position: 'absolute',
+      zIndex: 1,
+    },
+  }),
+);
+
+const LeftPanelWrapper = styled(PanelWrapper)(
+  ({ theme, open }: { theme: Theme; open: boolean }) => ({
+    borderRight: `1px solid ${theme.palette.cruGrayLight.main}`,
```
Comment (TheNoodleMoose) [inline src/components/Layouts/SidePanelsLayout.tsx:63]: Is there a difference between this gray and ```theme.palette.grey```? ```theme.palette.grey``` has a lot of options as well so I was curious if there was something specific about ```theme.palette.cruGrayLight.main```.

File: src/components/Layouts/SidePanelsLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+const PanelWrapper = styled(ScrollBox)(
+  ({
+    theme,
+    breakpoint,
+  }: {
+    theme: Theme;
+    breakpoint: Breakpoint | number;
+  }) => ({
+    flexGrow: 0,
+    flexShrink: 0,
```
[the remark this answers is not in the document]
Comment (OzzieOrca) [reply src/components/Layouts/SidePanelsLayout.tsx:51]: You know, grow might be defaulted to 0. Shrink isn't.
https://developer.mozilla.org/en-US/docs/Web/CSS/flex#formal_definition.

File: src/components/Layouts/SidePanelsLayout.tsx (hunk 1 above)
Comment (OzzieOrca) [reply src/components/Layouts/SidePanelsLayout.tsx:63]: #EBECEC (`cruGrayLight`) is a custom color from Cru's brand. It isn't in `theme.palette.grey`https://material-ui.com/customization/default-theme/.

Maybe there's an argument that we should customize the grey palette but it's got 14 greys in it. I'm not sure how you'd know which one to pick and design isn't customizing them.

**Comment by hotcakedev628** [comment]
This could be solved by implementing drawer for contact filters.
https://material-ui.com/components/drawers/#responsive-drawer
I think Task functionality already has the drawer.
Please let me know if you want me to work on this.

**Comment by OzzieOrca** [comment]
Finally figured out why the test `backgroundColor` was always `transparent` https://stackoverflow.com/a/66035529/665224. I've just disabled 3 tests for now...

Maybe a Material UI v5 upgrade and the switch from JSS to Emotion for CSS will fix something 🤷‍♂️ 

@hotcakedev628 The drawer component didn't have all the functionality I wanted. On large screens we want the side panels to slide in from the side and shrink the size of the other columns. On smaller screens we want the panels to slide in on top of the content. And on very small screens the contact details needs to take up all the width.

**Review by TheNoodleMoose — APPROVED** [review]

**Pull request merged**